tests/test_arrow_dataset.py

```diff
@@ -465,6 +465,45 @@ def test_cast(self, in_memory):
             del dset
             del casted_dset
 
+    def test_class_encode_column(self, in_memory):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            dset = self._create_dummy_dataset(in_memory, tmp_dir, multiple_columns=True)
+
+            with self.assertRaises(ValueError):
+                dset.class_encode_column(column="does not exist")
+
+            casted_dset = dset.class_encode_column("col_1")
+            self.assertIsInstance(casted_dset.features["col_1"], ClassLabel)
+            self.assertListEqual(casted_dset.features["col_1"].names, ["0", "1", "2", "3"])
+            self.assertListEqual(casted_dset["col_1"], [3, 2, 1, 0])
+            self.assertNotEqual(casted_dset._fingerprint, dset._fingerprint)
+            self.assertNotEqual(casted_dset, dset)
+            del casted_dset
```

Comment:
```suggestion
            with dset.class_encode_column("col_1") as casted_dset:
                self.assertIsInstance(casted_dset.features["col_1"], ClassLabel)
                self.assertListEqual(casted_dset.features["col_1"].names, ["0", "1", "2", "3"])
                self.assertListEqual(casted_dset["col_1"], [3, 2, 1, 0])
                self.assertNotEqual(casted_dset._fingerprint, dset._fingerprint)
                self.assertNotEqual(casted_dset, dset)
```
We try to avoid the use of `del` now (see #2191 ). Plus the `with` statement properly closes the dataset which makes the errors readable on windows.

Can you make this change to the 3 other cases you tested please ?